test/save_test.rb

```diff
@@ -41,6 +41,15 @@ def test_update
     @record.save
   end
 
+  def test_update_no_change
+    # Regular flow, write index id, write index id/tile, delete data blob since Record has changed
+    expect_cache_delete("#{NAMESPACE}attr:Item:id:id/title:#{cache_hash('"1"/"bob"')}").never
+    expect_cache_delete("#{NAMESPACE}attr:Item:id:title:#{cache_hash('"bob"')}").never
+    expect_cache_delete(@blob_key).never
```

Comment:
This type of test is quite fragile.  Not only is it coupled to the internals of the library, but also changes to those internals could cause this test to no longer test anything relevant by it instead deleting slightly different cache key.

A way to test this with less internal coupling would be to do a fetch surrounded by `assert_no_queries` to make sure the cache wasn't expired.